src/courier.js

```diff
@@ -63,38 +71,67 @@ const logToConsole = (level, origin, message) => {
   levelFormat[level](`[${time}] ${text}`)
 }
 
-const originMatch = (origin) => {
-  const {vendor, name, version} = manifest
-  const local = `${vendor}.${name}@${version.substring(0, 0)}`
-  return local === origin.substring(0, origin.indexOf('@') + 1)
+const appByMajor = (app) => {
+  const versionIndex = app.indexOf('@')
+  return versionIndex > -1
+    ? app.substring(0, versionIndex + 2) : app
 }
 
-const listen = (account, workspace, authToken) => {
-  let es = new EventSource(`${courierHost}/${account}/${workspace}/app-events?level=${log.level}`, {
+const getOriginByManifest = () => {
+  const {name, vendor, version} = manifest.name
+  return `${vendor}.${name}@${version}`
+}
+
+const originMatch = (msgOrigin, origin = []) => {
+  const msgOriginByMajor = appByMajor(msgOrigin)
+  if (origin.length === 0) {
+    origin.push(getOriginByManifest())
+  }
+  return any(app => msgOriginByMajor.indexOf(app) > -1, map(appByMajor, origin))
+}
+
+const listen = (account, workspace, authToken, {timeout: {duration, action}, origin, callback}) => {
```

Comment:
For the `install` and `uninstall` cases, no.
But I think that may exist other cases... The callback is for the `workspace:changed` event, specifically, while the timeout `action` is for any timeout that might happen.
I think it's possible someone wanting to handle these stuff differently.